advertools/spider.py

```diff
@@ -949,6 +949,14 @@ def crawl(url_list, output_file, follow_links=False,
             else:
                 setting = '='.join([key, str(val)])
             settings_list.extend(['-s', setting])
+    meta_list = []
+    if meta is not None:
+        for key, val in meta.items():
+            if isinstance(val, (dict, list, set, tuple)):
+                meta = '='.join([key, json.dumps(val)])
+            else:
+                meta = '='.join([key, str(val)])
+            meta_list.extend(['-s', meta])
```

Comment:
The `-s` flag is for the `custom_settings` parameter. I don't think we need it here. Unless I'm missing something?

Reply:
I think -s options scope is general, not specific to custom settings.

Please see below:

`Usage
=====
  scrapy runspider [options] <spider_file>

Run the spider defined in the given file

Optional Arguments
==================
  -h, --help            show this help message and exit
  -a NAME=VALUE         set spider argument (may be repeated)
  -o FILE, --output FILE
                        append scraped items to the end of FILE (use - for stdout)
  -O FILE, --overwrite-output FILE
                        dump scraped items into FILE, overwriting any existing file
  -t FORMAT, --output-format FORMAT
                        format to use for dumping items

Global Options
--------------
  --logfile FILE        log file. if omitted stderr will be used
  -L LEVEL, --loglevel LEVEL
                        log level (default: DEBUG)
  --nolog               disable logging completely
  --profile FILE        write python cProfile stats to FILE
  --pidfile FILE        write process ID to FILE
  -s NAME=VALUE, --set NAME=VALUE
                        set/override setting (may be repeated)
  --pdb                 enable pdb on failure`

Reply:
Trying to understand full call stack to understand e2e flow. 

Command in the runspider module is inheriting BaseRunSpiderCommand. Where options are being processed. 

https://github.com/scrapy/scrapy/blob/52d93490f57d4b2284e3d95efa5c622ab8ad16cb/scrapy/commands/runspider.py#L26

https://github.com/scrapy/scrapy/blob/52d93490f57d4b2284e3d95efa5c622ab8ad16cb/scrapy/commands/__init__.py#L109

What I have found so far is that scrapy runspider expects configurations in the supplied file (spider module) itself. Allowing to override settings using -s options.

It needs further investigation to see how args are handled within [self.crawler_process.crawl(spidercls, **opts.spargs)](https://github.com/scrapy/scrapy/blob/52d93490f57d4b2284e3d95efa5c622ab8ad16cb/scrapy/commands/runspider.py#L55)


Meanwhile, I have created an issue [here](https://github.com/scrapy/scrapy/issues/5610).

Reply:
We simply need to add an argument through the `-a` flag, and then that becomes an attribute of the spider, and can be used as a `meta` parameter wile making the request with `Request`. 

Again, are you sure we really need the meta parameter? 

Does the Playwright functionality work right now (without having to use `meta`)?

Reply:
>We simply need to add an argument through the -a flag, and then that becomes an attribute of the spider, and can be used as a meta parameter wile making the request with Request.

Actually yes! Last time we already discussed its implementation https://github.com/eliasdabbas/advertools/pull/207#discussion_r873175060

This time I got distracted looking into the logs. ;)

I will update the code accordingly. 

>Again, are you sure we really need the meta parameter?
In general, I am in the favor of having meta parameter as it allows us to set [Request.meta](https://docs.scrapy.org/en/latest/topics/request-response.html#scrapy.http.Request.meta) per request for different Scrapy components (extensions, middlewares, etc). And it can also be accessed in response.meta or callback that handles the request.

Also for the playwright, we still need it:

`import scrapy

class AwesomeSpider(scrapy.Spider):
    name = "awesome"

    def start_requests(self):
        # GET request
        yield scrapy.Request("https://httpbin.org/get", meta={"playwright": True})
        # POST request
        yield scrapy.FormRequest(
            url="https://httpbin.org/post",
            formdata={"foo": "bar"},
            meta={"playwright": True},
        )

    def parse(self, response):
        # 'response' contains the page as seen by the browser
        yield {"url": response.url}`

Reply:
Another suggestion is to create a dedicated spider for playwright.

Reply:
>  Actually yes! Last time we already discussed its implementation https://github.com/eliasdabbas/advertools/pull/207#discussion_r873175060

So why did you ignore that?

>     yield scrapy.FormRequest(
>        url="https://httpbin.org/post",
>       formdata={"foo": "bar"},

This is a `FormRequest` for filling out forms on pages, and has nothing to do with this spider, which runs a normal `Request`.

> It will make sense to have a separate spider as certain use cases require handling response parsring differently.

Why did you create it with the regular spider then? 

Please make sure you run through the documentation carefully, and make sure that it is clear why we are introducing a new change before submitting a PR, and that it is working.